tabbycat/results/scoresheet.py

```diff
@@ -72,93 +73,102 @@ def identical(self, other):
+        winners_declared = self.declared_winners.issubset(set(self.sides)) and len(self.declared_winners) == self.number_winners
+        return super().is_complete() and winners_declared
 
-    def set_declared_winner(self, winner):
-        assert winner in self.sides or winner is None, "Declared winner must be one of " + ", ".join(map(repr, self.sides))
-        self.declared_winner = winner
+    def add_declared_winner(self, winner):
+        assert winner in self.sides or winner is None, "Declared winner must be one of: " + ", ".join(map(repr, self.sides))
+        self.declared_winners.add(winner)
 
-    def get_declared_winner(self):
-        return self.declared_winner
+    def set_declared_winners(self, winners):
+        winners = set(winners)
+        assert winners.issubset(set(self.sides)) or len(winners) == 0, "Declared winners must be in: " + ", ".join(map(repr, self.sides))
+        self.declared_winners = winners
 
     def identical(self, other):
-        return super().identical(other) and self.declared_winner == other.declared_winner
+        return super().identical(other) and set(self.declared_winners) == set(other.declared_winners)
+
+    def _get_winners(self):
+        assert len(self.declared_winners) == self.number_winners, "There can only be this number of winners: %d" % self.number_winners
+        return self.declared_winners
 
 
 class BaseTwoTeamScoresheet(BaseScoresheet):
 
     sides = ['aff', 'neg']
-
-    def winner(self):
-        """Returns 'aff' is the affirmative team won, and 'neg' if the negative
-        team won. `self._get_winner()` must be implemented by subclasses."""
-        if not self.is_complete():
-            return None
-        return self._get_winner()
+    number_winners = 1
 
     def is_valid(self):
-        return super().is_valid() and self.winner() is not None
+        return super().is_valid() and len(self.winners()) == 1
 
+    def rank(self, side):
+        rank = 1 if side in self.winners() else 2
+        return rank
 
-class ResultOnlyScoresheet(DeclaredWinnerMixin, BaseTwoTeamScoresheet):
-    """Winner only, no scores. In this case the scoresheet is basically just
-    a shell for a single piece of data (the winner)."""
 
-    def _get_winner(self):
-        return self.declared_winner
+class ResultOnlyScoresheet(DeclaredWinnersMixin, BaseTwoTeamScoresheet):
+    """Winners only, no scores. In this case the scoresheet is basically just
+    a shell for a single piece of data (the winners)."""
+    pass
 
 
 class HighPointWinsRequiredScoresheet(ScoresMixin, BaseTwoTeamScoresheet):
     """Draws are not permitted; winning teams must have a higher total.
     This is the standard type of scoresheet in Asia and Oceania."""
 
-    def _get_winner(self):
+    def _get_winners(self):
         aff_total = self.get_total('aff')
         neg_total = self.get_total('neg')
         if aff_total > neg_total:
-            return 'aff'
+            return set(['aff'])
```

Comment:
Not a showstopper but if you like, the syntax `{'aff'}` does the same thing 🙂

(So long as there's no colon, it's a set not a dict.)